File: TP1/Exo 1.6/python/triangle_while.py
def triangle(taille):
    # Validation pour vérifier que la taille est strictement inférieure à 10
    if taille >= 10:
        print("La taille doit être strictement inférieure à 10.")
        return  # Sortie de la fonction si la taille n'est pas valide
    
    i = 1  # Initialisation de la variable de contrôle pour la boucle des lignes

    # Boucle pour dessiner le triangle
    while i <= taille:  # Parcourt chaque ligne du triangle
        line = ""  # Initialise la chaîne de caractères pour la ligne courante
        j = 0  # Initialisation de la variable de contrôle pour la boucle des colonnes
        while j < i:  # Parcourt chaque position dans la ligne courante
            # Affichage des contours avec '*'
            if j == 0 or j == i - 1 or i == taille:  # Premier ou dernier caractère de la ligne, ou si c'est la dernière ligne
                line += "* "  # Ajoute '*' à la chaîne de caractères
            else:
                # Affichage de l'intérieur avec '#'
                line += "# "  # Ajoute '#' pour les positions intérieures
            j += 1  # Incrémente la variable de colonne
        print(line.strip())  # Affiche la ligne courante en retirant les espaces supplémentaires à la fin
        i += 1  # Incrémente la variable de ligne

def main():
    # Demande à l'utilisateur d'entrer une taille pour le triangle
    taille = int(input("Entrez une valeur de taille de triangle inférieure à 10 : "))
    triangle(taille)  # Appelle la fonction pour dessiner le triangle

# Point d'entrée du programme
if __name__ == "__main__":
    main()

File: TP1/Exo 1.6/C/triangle_while.c
#include <stdio.h> // Inclusion de la bibliothèque standard pour les entrées/sorties

// Fonction pour dessiner un triangle
void triangle(int taille)
{
    int i = 1;  // Initialisation de la variable de contrôle de la boucle pour les lignes

    // Boucle pour chaque ligne du triangle, de 1 à la taille donnée
    while (i <= taille)
    {
        int j = 0;  // Initialisation de la variable de contrôle de la boucle pour les colonnes
        // Boucle pour chaque caractère à afficher sur la ligne i
        while (j < i)
        {
            // Conditions pour savoir quand afficher un "*" ou un "#"
            if (j == 0 || j == i - 1 || i == taille)
            {
                printf("* ");  // Affiche "*" si on est sur la bordure du triangle ou sur la dernière ligne
            }
            else
            {
                printf("# ");  // Affiche "#" si on est à l'intérieur du triangle, sans bordure
            }
            j++;  // Incrémente la variable de colonne
        }
        printf("\n");  // Nouvelle ligne pour passer à la ligne suivante du triangle
        i++;  // Incrémente la variable de ligne
    }
}

int main()
{
    int taille;

    // Demande à l'utilisateur de saisir une taille pour le triangle
    printf("Entrez une valeur pour la taille (strictement inférieure à 10) : ");
    // Vérifie que l'utilisateur entre un entier valide entre 1 et 9
    while (scanf("%d", &taille) != 1 || taille >= 10 || taille < 1) {
        printf("Veuillez entrer un entier valide entre 1 et 9 : ");
        while (getchar() != '\n'); // Vide le buffer d'entrée en cas d'entrée non valide
    }

    // Appelle la fonction triangle pour dessiner le triangle de la taille donnée
    triangle(taille);

    return 0; // Fin du programme
}
